tools/initialize.py

```diff
@@ -0,0 +1,404 @@
+from convert_partition import gen_dgl_objs
+
+def send_node_data(rank, node_data, part_list):
+    """
+    Function to send node_data to the non-rank-0 processes. 
+
+    Parameters:
+    -----------
+    rank : rank of the process
+    node_data : node_data, in the augmented form
+    ranks_list : list of unique ranks/partition-ids
+    """
+
+    for part_id in part_list: 
+        if part_id == rank: 
+            continue
+        
+        #extract <node_type>, <orig_type_id>, <global_id>
+        #which belong to `part_id`
+        send_data = (node_data[:, 7] == part_id) 
+        idx = send_data.reshape(node_data.shape[0])
+        filtered_send_data = node_data[:,[0,5,6]][idx == 1] # send node_type, orig_type_id, line_id (global_id)
+
+        #
+        send_size = filtered_send_data.shape
+        send_tensor = torch.zeros(len(send_size), dtype=torch.int)
+        for idx in range(len(send_size)): 
+            send_tensor[idx] = send_size[idx]
+
+        # Send size first, so that the part-id (rank)
+        # can create appropriately sized buffers
+        dist.send(send_tensor, dst=part_id)
+
+        #send actual node_data to part-id rank
+        start = timer ()
+        send_tensor = torch.from_numpy( filtered_send_data.astype(np.int32) )
+        dist.send(send_tensor, dst=part_id)
+        end = timer ()
+        print( 'Rank: ', rank, ' Sent data size: ', filtered_send_data.shape, ', to Process: ', part_id, 'in: ', timedelta(seconds = end - start) )
+
+
+def send_edge_data(rank, edge_data, rank_list): 
+    """
+    Function to send edge data to non-rank-0 processes
+
+    Parameters:
+    -----------
+    rank : rank of the process
+    node_data : node_data, in the augmented form
+    ranks_list : list of unique ranks/partition-ids
+    """
+    for part_id in rank_list: 
+        if part_id == rank: 
+            continue
+
+        send_data = (edge_data[:, 4] == part_id) 
+        idx = send_data.reshape(edge_data.shape[0])
+        filtered_send_data = edge_data[:,[0,1,2,3]][idx == 1]
+
+        send_size = filtered_send_data.shape
+        send_tensor = torch.zeros(len(send_size), dtype=torch.int32)
+        for idx in range(len(send_size)): 
+            send_tensor[idx] = send_size[idx]
+
+        # Send size first, so that the rProc can create appropriately sized tensor
+        dist.send(send_tensor, dst=part_id)
+
+        start = timer()
+        send_tensor = torch.from_numpy(filtered_send_data.astype(np.int32))
+        dist.send(send_tensor, dst=part_id)
+        end = timer()
+
+        print('Rank: ', rank, ' Time to send Edges to proc: ', part_id, ' is : ', timedelta(seconds = end - start))
+
+
+def send_node_features(rank, node_data, node_features, rank_list, ntype_map):
+    """
+    Function to send node_features data to non-rank-0 processes.
+
+    Parameters:
+    -----------
+    rank : rank of the process
+    node_features : node_features, data from the node_feats.dgl
```

Comment:
i don't understand. i thought `node_features` are the node features loaded from users input. why is it loaded from `node_feats.dgl`?